assets/src/application/ui/calendars/dateDisplay/EditDateRangeButton.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useCallback } from 'react';
+import { useDisclosure } from '@chakra-ui/hooks';
+import { __ } from '@wordpress/i18n';
+
+import { DateTimeRangePicker } from '../../input';
+import { ButtonSize, ButtonType, IconButton } from '../../input';
+import { CalendarOutlined, Popover } from '../../display';
+import { EditDateButtonProps } from './types';
+
+const EditDateRangeButton: React.FC<EditDateButtonProps> = ({ header, onEditHandler, startDate, endDate, tooltip }) => {
+	const { isOpen, onOpen, onClose } = useDisclosure();
+	const headerText = header ? header : __('Edit Start and End Dates and Times');
+	const onChange = useCallback(
+		(dates: string[]) => {
+			onEditHandler(dates);
+			onClose();
+		},
+		[onClose, onEditHandler]
+	);
+	return (
+		<Popover
+			className={'ee-edit-calendar-date-range'}
+			closeOnBlur={false}
+			content={<DateTimeRangePicker endDate={endDate} startDate={startDate} onChange={onChange} />}
+			header={<strong>{headerText}</strong>}
+			isOpen={isOpen}
+			onClose={onClose}
+			trigger={
+				<IconButton
+					borderless
+					buttonSize={ButtonSize.SMALL}
+					buttonType={ButtonType.MINIMAL}
+					className={'ee-edit-calendar-date-range-btn'}
+					color={'white'}
+					onClick={onOpen}
+					tooltip={tooltip}
+					icon={CalendarOutlined}
```

Comment:
almost purfect

Reply:
LOLZ... well... what part can be improved? 😄

Reply:
the alphabetical order, but this is really nitpicky, so maybe leave as is :)